hd-wallet/pluginv2.lua

```diff
@@ -0,0 +1,511 @@
+    data = data..index_hex
+    hmac = get_hmac(parent_key.chain_code, data)
+
+    child_key = {
+        index = index_hex,
+        chain_code = string.sub(hmac, 65, 128),
+        depth = num_2_hex(tonumber(parent_key.depth + 1), 2),
+        is_private = parent_key.is_private,
+    }
+  
+    if parent_key.version == PRIVATE_WALLET_VERSION then
+        child_key.version = PRIVATE_WALLET_VERSION 
+        fingerprint = hash160(public_key_for_private_key(string.sub(parent_key.key, 3, 66))) 
+        child_key.fingerprint = string.sub(fingerprint, 1, 8)
+        -- appending 00 to make key size 33 bit
+        child_key.key = "00"..tostring(add_private_keys(string.sub(hmac, 1, 64), parent_key.key))
+    else
+        child_key.version = PUBLIC_WALLET_VERSION
+        fingerprint = hash160(parent_key.key)
+        child_key.fingerprint = string.sub(fingerprint, 1, 8)
+        key_bytes = public_key_for_private_key(string.sub(hmac, 1, 64))
+        child_key.key = add_public_keys(key_bytes, parent_key.key)
+    end
+    
+   child_key.checksum = string.sub(get_check_sum(child_key), 1, 8)
+  
+    return child_key
+end
+-- serialize child key and encode into base58
+function serialize(child_key)
+    local child_key_string = table.concat({child_key.version, child_key.depth, child_key.fingerprint, child_key.index, 
+        child_key.chain_code, child_key.key, child_key.checksum}
+    )
+    local blob = Blob.from_hex(child_key_string)
+    return  blob:base58()
+end
+----------- @@@@@@@@@@@@@@@@@@@@@@@@@ -----------
+-----------            ETH            -----------
+----------- @@@@@@@@@@@@@@@@@@@@@@@@@ -----------
+function get_rs(signature)
+  local signature_length = tonumber(string.sub(signature, 3, 4), 16) + 2
+
+  local r_length = tonumber(string.sub(signature, 7, 8), 16)
+  local r_left = 9
+  local r_right = r_length*2 + r_left - 1
+  local r = string.sub(signature, r_left, r_right)
+  local s_left = r_right + 5
+  local s_right = signature_length*2
+  local s_length = tonumber(string.sub(signature, s_left-2, s_left-1), 16)
+  local s = string.sub(signature, s_left, s_right)
+
+  -- If s is negative, change it to (-s:mod(N))
+  local sign = tonumber(string.sub(s, 1, 2), 16)
+  if (sign > 127) then
+    s = BigNum.from_bytes_be(~Blob.from_hex(s))
+    s:add(BigNum.from_bytes_be(Blob.from_hex("01")))
+    s:mod(BigNum.from_bytes_be(Blob.from_hex(N)))
+    s = s:hex()
+  end
+
+  return {
+    r = r,
+    s = s
+  }
+end
+function set_length_left(msg, len)
+  local msg_len = string.len(msg)
+  if (msg_len > len*2) then
+    -- This one's too big: truncate it
+    return string.sub(msg, msg_len - len*2 + 1)
+  else
+    -- Otherwise: pad it
+    return string.rep("0", len*2 - msg_len)..msg
+  end
+end
+function get_eth_v(r,s)
+  local a = BigNum.from_bytes_be(Blob.from_hex(r))
+  local b = BigNum.from_bytes_be(Blob.from_hex("02"))
+  a:mod(b)
+  if a:to_bytes_be()[1] == "" then
+    return "1B"
+  else
```

Comment:
There is a corner case here where on certain r and s values this fails.  Consider checking for this case